includes/admin/payments/view-payment-details.php

```diff
@@ -384,7 +384,7 @@
 															'name'             => 'give-variable-price',
 															'chosen'           => true,
 															'show_option_all'  => '',
-															'show_option_none' => '',
+															'show_option_none' => ( '' === get_post_meta( $payment_id, '_give_payment_price_id', true ) ? __( 'None', 'give' )  : '' ),
```

Comment:
@ravinderk @mehul0810 anything that doesn't match a donation level or set donation price should be marked as `custom` there should be no blank option